doc/source/train/user-guides/experiment-tracking.rst

```diff
@@ -1,195 +1,255 @@
+``train_loop_per_worker`` needs to have access to the tracking service
+----------------------------------------------------------------------
 
-        1. Your code is executed in parallel on many workers. However, you often only want to report
-        results from one of these workers (usually the first worker - the "rank 0" worker).
+Usually this is trivial for online externally hosted tracking services. 
+Other than that, sometimes offline mode is easier to configure. 
+For example, for MLflow, you can use file based tracking uri, 
+where the uri points to a shared storage location. For Tensorboard, 
+you can have SummaryWriter point to a shared storage location as well.
 
-        2. When using the native libraries, you should report the results to both Ray Train and
-        the experiment tracking library.
+Advanced usage:
+Shared storage location for offline mode may not be necessary. 
+If you prefer to not use shared storage in this case, 
+you can log to local and have the logged file transferred to the head node 
+or to the cloud post training. 
 
-        Example:
+``train_loop_per_worker`` needs to have access to credentials
+-------------------------------------------------------------
 
-        .. code-block:: python
+There are mainly two mechanisms when accessing credentials: 
+checking some env var value, and checking the content of some config file. 
+When applicable, you need to make sure that env var is set in the training 
+function and the file is stored on shared storage to be accessible.
+
+Avoid duplications
+------------------
+
+There are probably multiple ``train_loop_per_worker`` processes running, 
+each corresponding to a Ray Train Worker. For distributed data 
+parallel training, all of them should have the same results. 
+You may want to only log from rank 0 worker to avoid duplications. 
+You may find how to do it in the following code examples.
+
+
+Code Example
+============
+
+In the following session, we will show some examples using Wandb 
+and MLflow but the idea should be easily adaptable to other frameworks.
+
+Pytorch
+-------
+
+Conceptual code snippets
+~~~~~~~~~~~~~~~~~~~~~~~~
 
+.. tabs::
+
+    .. tab:: wandb
+
+        .. code-block:: python
+            
             from ray import train
+            import wandb
 
-            def train_fn(config):
-                context = train.get_context()
+            # assuming passing API key through config
+            def train_func(config):
+                os.environ["WANDB_API_KEY"] = config["wandb_api_key"]
 
                 wandb.init(
-                    id=context.get_trial_id(),
-                    name=context.get_trial_name(),
-                    group=context.get_experiment_name(),
+                    id=..., # or train.get_context().get_trial_id()
+                    name=..., # or train.get_context().get_trial_name(),
+                    group=..., # or train.get_context().get_experiment_name(),
```

Comment:
I would suggest moving the out, or making it clearer that these are suggested values.

Reply:
reworked the entire session above to be "steps" oriented instead of anchoring on dist vs non dist. Please take a look